test/models/book_test.rb

```diff
@@ -0,0 +1,33 @@
+require 'test_helper'
+
+class BookTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+
+  test "Books must have a title, author, genre and rank" do
+    assert books(:red_fern).valid? "Red Fern Grows should be valid"
+
+    books(:red_fern).title = nil  ## This name removal is local to the test data - it doesn't change the fixture
+    assert_not books(:red_fern).valid? "Red Fern Grows should no longer be valid"
+
+    books(:dingo).rank = nil
+    assert_not books(:dingo).valid? "Dingo should no longer be valid"
+  end
+
+
+  test "Books must have a unique combination of title and author" do
+    assert books(:red_fern_title).valid? "Red Fern with same title and different author should be valid"
+
+    assert_not Book.new(title: "Where the Red Fern Grows", author: "Wilson Rawls", genre: "Children", description: "About a boy who buys and trains two Redbone Coonhound hunting dogs", rank: 10).valid? "Red Fern with same title and same author should not be valid"
+  end
+
+  test 'Books can be upvotes or downvote on' do
+    books(:red_fern).upvote
+    assert_equal books(:red_fern).rank, 55
+
+    books(:red_fern).downvote
+    assert_equal books(:red_fern).rank, 54
```

Comment:
To make these tests more general we should store the original value of `rank` before calling `upvote`\`downvote` and then assert that the new value is +1 or -1 as appropriate.